soc/arm/nxp_imx/rt/soc_rt11xx.c

```diff
@@ -624,36 +625,9 @@ static int imxrt_init(const struct device *arg)
 #if defined(CONFIG_SOC_MIMXRT1176_CM4) || defined(CONFIG_SOC_MIMXRT1166_CM4)
 	/* Initialize Cache */
 	/* Enable Code Bus Cache */
-	if (0U == (LMEM->PCCCR & LMEM_PCCCR_ENCACHE_MASK)) {
-		/*
-		 * set command to invalidate all ways,
-		 * and write GO bit to initiate command
-		 */
-		LMEM->PCCCR |= LMEM_PCCCR_INVW1_MASK
-			| LMEM_PCCCR_INVW0_MASK | LMEM_PCCCR_GO_MASK;
-		/* Wait until the command completes */
-		while ((LMEM->PCCCR & LMEM_PCCCR_GO_MASK) != 0U) {
-		}
-		/* Enable cache, enable write buffer */
-		LMEM->PCCCR |= (LMEM_PCCCR_ENWRBUF_MASK
-				| LMEM_PCCCR_ENCACHE_MASK);
-	}
-
+	cache_instr_enable();
```

Comment:
where all the code you removed ended up to?

Reply:
it was unnecessary because there is an NXP HAL function for it